File: app/src/main/java/com/example/myapp/History/FragmentHistory.java
package com.example.myapp.History;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.navigation.fragment.NavHostFragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.myapp.MainApp;
import com.example.myapp.R;

import java.util.ArrayList;
import java.util.List;

import static com.example.myapp.Common.Utils.LOGCAT_TAG;

public class FragmentHistory extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater,
                             ViewGroup container,
                             Bundle savedInstanceState) {
        setHasOptionsMenu(true);
        return inflater.inflate(R.layout.fragment_history, container, false);
    }

    @Override
    public void onCreateOptionsMenu(@NonNull Menu menu, @NonNull MenuInflater inflater) {
        super.onCreateOptionsMenu(menu, inflater);
        menu.findItem(R.id.action_search).setVisible(false);
        menu.findItem(R.id.action_my_location).setVisible(false);
        menu.findItem(R.id.action_favorite).setVisible(false);
        menu.findItem(R.id.action_start).setVisible(true);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == R.id.action_start) {
            NavHostFragment.findNavController(this).navigate(R.id.actionStart, null);
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        getActivity().setTitle(R.string.label_history);
        RecyclerView rvHistory =  view.findViewById(R.id.rvHistory);
        rvHistory.setAdapter(new AdapterHistory());
    }
}

File: app/src/main/java/com/chplalex/shaman/ui/fragment/FragmentHistory.kt
package com.chplalex.shaman.ui.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.chplalex.shaman.ui.adapter.AdapterHistory
import com.chplalex.shaman.R
import com.chplalex.shaman.mvp.presenter.PresenterAbout
import com.chplalex.shaman.utils.showToast
import com.chplalex.shaman.mvp.presenter.PresenterHistory
import com.chplalex.shaman.mvp.view.IViewHistory
import com.chplalex.shaman.ui.App
import com.chplalex.shaman.ui.App.Companion.instance
import moxy.MvpAppCompatFragment
import moxy.ktx.moxyPresenter
import javax.inject.Inject
import javax.inject.Provider

class FragmentHistory : MvpAppCompatFragment(R.layout.fragment_history), IViewHistory {

    @Inject
    lateinit var injectPresenter: Provider<PresenterHistory>

    private val presenter by moxyPresenter {
        injectPresenter.get()
    }

    private val adapter by lazy {
        AdapterHistory(presenter.presenterList)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        instance.activityComponent?.inject(this)
        super.onCreate(savedInstanceState)
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        super.onCreateOptionsMenu(menu, inflater)
        inflater.inflate(R.menu.menu_no_start, menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.action_start) {
            presenter.onActionStart()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        activity?.setTitle(R.string.label_history)
        setHasOptionsMenu(true)
        view.findViewById<RecyclerView>(R.id.rvHistory).adapter = adapter
    }

    override fun showErrorDB(error: Throwable) {
        context?.showToast("Ошибка БД: $error")
    }

    override fun notifyItemRemoved(pos: Int) = adapter.notifyItemRemoved(pos)
    override fun notifyItemChanged(pos: Int) = adapter.notifyItemChanged(pos)
    override fun notifyDataSetChanged() = adapter.notifyDataSetChanged()
}